src/app/(account)/join-us/ProgressBar.tsx

```diff
@@ -1,31 +1,32 @@
 import Image from 'next/image';
 
-interface ProgressBarProps {
-    ducksFilled: number;
-}
-
-function ProgressBar({ ducksFilled }: ProgressBarProps) {
-    const renderDucks = () => {
-        const ducks = [];
-        for (let i = 1; i <= 4; i++) {
-            const duckColor = i <= ducksFilled ? 'yellowDuck.svg' : 'greyDuckOutline.svg';
-            ducks.push(
-                <div key={i} className="flex items-center justify-center">
-                    <Image
-                        src={`/images/${duckColor}`}
-                        alt={i <= ducksFilled ? 'Yellow Duck' : 'Grey Duck Outline'}
-                        className="h-10 md:h-12 scale-x-[-1]"
-                        height={100}
-                        width={100}
-                    />
-                    <div className="absolute mt-20 text-bla ck font-bold">{i}</div>
-                </div>
-            );
-        }
-        return ducks;
-    };
+function Duck({ filled, index }: { filled?: boolean; index: number }) {
+    const duckImageName = filled ? 'yellowDuck.svg' : 'greyDuckOutline.svg';
+    const duckImageAlt = filled ? 'Yellow Duck' : 'Grey Duck Outline';
 
-    return <div className="flex items-end justify-center mt-4 mb-12">{renderDucks()}</div>;
+    return (
+        <div className="flex items-center justify-center">
+            <Image
+                src={`/images/${duckImageName}`}
+                alt={duckImageAlt}
+                className="h-10 md:h-12 scale-x-[-1]"
+                height={100}
+                width={100}
+            />
+            <div className="absolute mt-20 text-black font-bold">{index}</div>
+        </div>
+    );
 }
 
-export default ProgressBar;
+export default function ProgressBar({ step }: { step: number }) {
+    return (
+        <div className="flex items-end justify-center mt-4 mb-12">
+            {new Array(step).fill(null).map((_, i) => (
+                <Duck filled index={i + 1} key={i} />
+            ))}
+            {new Array(4 - step).fill(null).map((_, i) => (
+                <Duck index={step + i + 1} key={i} />
+            ))}
+        </div>
+    );
+}
```

Comment:
- IMHO there is no way to make `step` greater than 4. Only way to control `step` is the `continue` button in each step. In the step 4, there isn't such a button. So I assume a check is unnecessary.
- Yea I also think it's very convoluted. But I assume it's the only way to make inline `for` loop in jsx in react. I'd like to know if you have any better suggestions.

Reply:
Yeah I think that convoluted method is the only way to use a for loop within the JSX element itself. It's possible to extract out the logic and append things to an array, then cycle over it in the JSX element using a map once again. Whether this will be simpler than the current method, I'm not too sure. I guess it's fine as it is for the time being.

If the only way to make `step` greater than 4 is through the button then that's fine, no bounds checking is probably needed.

Reply:
yea right. But IMO `for` loop (extract out the logic) is for a more complex component. I want to keep this one simple by using a inline form.

Reply:
> But IMO `for` loop (extract out the logic) is for a more complex component.

I agree